packages/docusaurus-plugin-client-redirects/src/writeRedirectFiles.ts

```diff
@@ -42,7 +42,7 @@ export function toRedirectFilesMetadata(
     );
     const toUrl = addTrailingSlash(
       `${removeTrailingSlash(pluginContext.baseUrl)}${path.join(redirect.to)}`,
-    );
+    ).replace(/\\/g, '/');
```

Comment:
Hi and thanks for working on this issue.

I have a hard time understanding where the '\\' comes from, can you log the baseurl and redirect.to params so that we figure this out?

I feel like your solution works but is more a quickfix, and we should rather fix the problem at its root, where the \ is used instead of /

Reply:
Can you try running this code on your computer?

```
const path = require("path");

function addTrailingSlash(str) {
  return str.endsWith('/') ? str : `${str}/`;
}

function removeTrailingSlash(str) {
  return removeSuffix(str, '/');
}

function removeSuffix(str, suffix) {
  if (suffix === '') {
    return str; // always returns "" otherwise!
  }
  return str.endsWith(suffix) ? str.slice(0, -suffix.length) : str;
}


const baseUrl = "/";
const to = '/docs/something/else';



const toUrl = addTrailingSlash(
  `${removeTrailingSlash(baseUrl)}${path.join(to)}`
);

console.log(toUrl);
```

I suspect the `path.join(to)` is what adds the `\\`, it does not seem correct to use it to construct a URL path.

We could replace it with the `normalizeUrl([baseUrl,to])` in `@docusaurus/utils`

Reply:
Should be able to get this fixed thanks to CI and windows tests, FYI I'm pushing to your branch so that you are credited for your contribution

Reply:
oh, unfortunately forgot that our Jest setup on windows is still not good 😓 (https://github.com/facebook/docusaurus/pull/3540)

I believe this "normalizeUrl" could fix the issue, so going to merge but please let me know if you still have the problem after this change I made to your PR

Reply:
@slorber  Glad to help.  

The `\` is defiantly coming from the `path.join(to)`.  Windows is special and uses `\` for it's path separator unlike any other OS.

Here are the results from the test code above.

On Windows, I get `\docs\something\else/`.  
On WSL I get the expected output `/docs/something/else/`.  

I do agree that `path.join(to)` is probably the best way to to construct URLs, but I did find `path.posix.join(to)` produces the expected output on Windows.